analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -55,6 +61,43 @@ abstract class PackageManager {
                     // TODO: Bundler
             )
         }
+
+        /**
+         * Recursively search for files managed by a package manager.
+         *
+         * @param directory The root directory to search for managed files.
+         * @param packageManagers A list of package managers to use, defaults to [ALL].
+         */
+        fun findManagedFiles(directory: File, packageManagers: List<PackageManagerFactory<PackageManager>> = ALL)
```

Comment:
Do we need to define the `packageManagers`'s type here?

Reply:
Yes.